storops/unity/resource/host.py

```diff
@@ -74,6 +76,80 @@ def get_host(cls, cli, _id, force_create=False):
             ret = cls.get(cli=cli, _id=_id)
         return ret
 
+    def detach_hlu(self, lun):
+        return lun.detach_from(self)
+
+    def has_hlu(self, lun):
+        if not self.host_luns or not lun:
+            return False
+
+        has = False
+        for host_lun in self.host_luns:
+            if host_lun.lun is not None:
+                if host_lun.lun.id == lun.id:
+                    has = True
+                    break
+
+        return has
+
+    def get_hlu(self, lun):
+        if not self.host_luns or not lun:
+            return None
+
+        for host_lun in self.host_luns:
+            if host_lun.lun is not None:
+                if host_lun.lun.id == lun.id:
+                    hlu = host_lun.hlu
+                    break
+        else:
+            log.debug('Lun {} is not attached to host {}'
+                      .format(lun.name, self.name))
+            # TODO: raise UnityLunNotAttach?
+            hlu = None
+        return hlu
+
+    def add_initiator(self, name, type, force_create=True, **kwargs):
+        initiators = UnityHostInitiatorList.get(cli=self._cli,
+                                                initiator_id=name)
+
+        # Even if no initiators are found, the initiators object still contain
+        # one fake initiator.
```

Comment:
Yes.  I should have add more document about this.  For unity, if the filter is name or id, the return value should be a single instance for ease of use.  The original idea is that I could save duplicate APIs for getting single instance.  For other filters, they return list because we have no guarantee the result is a single instance.